providers/implementations/signature/eddsa_sig.c

```diff
@@ -589,47 +874,147 @@ static const OSSL_PARAM *eddsa_settable_ctx_params(ossl_unused void *vpeddsactx,
     return settable_ctx_params;
 }
 
-const OSSL_DISPATCH ossl_ed25519_signature_functions[] = {
-    { OSSL_FUNC_SIGNATURE_NEWCTX, (void (*)(void))eddsa_newctx },
-    { OSSL_FUNC_SIGNATURE_DIGEST_SIGN_INIT,
-      (void (*)(void))eddsa_digest_signverify_init },
-    { OSSL_FUNC_SIGNATURE_DIGEST_SIGN,
-      (void (*)(void))ed25519_digest_sign },
-    { OSSL_FUNC_SIGNATURE_DIGEST_VERIFY_INIT,
-      (void (*)(void))eddsa_digest_signverify_init },
-    { OSSL_FUNC_SIGNATURE_DIGEST_VERIFY,
-      (void (*)(void))ed25519_digest_verify },
-    { OSSL_FUNC_SIGNATURE_FREECTX, (void (*)(void))eddsa_freectx },
-    { OSSL_FUNC_SIGNATURE_DUPCTX, (void (*)(void))eddsa_dupctx },
-    { OSSL_FUNC_SIGNATURE_GET_CTX_PARAMS, (void (*)(void))eddsa_get_ctx_params },
-    { OSSL_FUNC_SIGNATURE_GETTABLE_CTX_PARAMS,
-      (void (*)(void))eddsa_gettable_ctx_params },
-    { OSSL_FUNC_SIGNATURE_SET_CTX_PARAMS, (void (*)(void))eddsa_set_ctx_params },
-    { OSSL_FUNC_SIGNATURE_SETTABLE_CTX_PARAMS,
-      (void (*)(void))eddsa_settable_ctx_params },
-    OSSL_DISPATCH_END
+static const OSSL_PARAM settable_variant_ctx_params[] = {
+    OSSL_PARAM_octet_string(OSSL_SIGNATURE_PARAM_CONTEXT_STRING, NULL, 0),
+    OSSL_PARAM_END
 };
 
-const OSSL_DISPATCH ossl_ed448_signature_functions[] = {
-    { OSSL_FUNC_SIGNATURE_NEWCTX, (void (*)(void))eddsa_newctx },
-    { OSSL_FUNC_SIGNATURE_DIGEST_SIGN_INIT,
-      (void (*)(void))eddsa_digest_signverify_init },
-    { OSSL_FUNC_SIGNATURE_DIGEST_SIGN,
-      (void (*)(void))ed448_digest_sign },
-    { OSSL_FUNC_SIGNATURE_DIGEST_VERIFY_INIT,
-      (void (*)(void))eddsa_digest_signverify_init },
-    { OSSL_FUNC_SIGNATURE_DIGEST_VERIFY,
-      (void (*)(void))ed448_digest_verify },
-    { OSSL_FUNC_SIGNATURE_FREECTX, (void (*)(void))eddsa_freectx },
-    { OSSL_FUNC_SIGNATURE_DUPCTX, (void (*)(void))eddsa_dupctx },
-    { OSSL_FUNC_SIGNATURE_GET_CTX_PARAMS, (void (*)(void))eddsa_get_ctx_params },
-    { OSSL_FUNC_SIGNATURE_GETTABLE_CTX_PARAMS,
-      (void (*)(void))eddsa_gettable_ctx_params },
-    { OSSL_FUNC_SIGNATURE_SET_CTX_PARAMS, (void (*)(void))eddsa_set_ctx_params },
-    { OSSL_FUNC_SIGNATURE_SETTABLE_CTX_PARAMS,
-      (void (*)(void))eddsa_settable_ctx_params },
+static const OSSL_PARAM *
+eddsa_settable_variant_ctx_params(ossl_unused void *vpeddsactx,
+                                  ossl_unused void *provctx)
+{
+    return settable_variant_ctx_params;
+}
+
+/*
+ * Ed25519 can be used using:
```

Comment:
```suggestion
 * Ed25519 can be used via:
```
Or perhaps `with` if you don't like `via`.

Reply:
I chose 'with'